ui/plugins/pinboard.js

```diff
@@ -184,6 +184,13 @@ treeherder.controller('PinboardCtrl', [
                    (thisClass.failure_classification_id === 2 && thisClass.text.length > 7));
         };
 
+        // Facilitates Clear all if no jobs pinned to reset pinboard UI
+        $scope.pinboardIsDirty = function () {
+            return $scope.classification.text !== '' ||
+                   thPinboard.hasRelatedBugs() ||
+                   $scope.classification.failure_classification_id !== 4;
```

Comment:
So I didn't bother with checking for $dirty on the select option, I think this is simpler and it accomplishes exactly the same thing. If we changed the default drop down from intermittent sure, we'd need to adjust this. But I don't see that as any more brittle than all the other tests we have in `canSaveClassifications()` and probably elsewhere.